src/platform/plugins/shared/discover/public/application/main/components/layout/cascaded_documents/blocks/cascade_leaf_component.tsx

```diff
@@ -0,0 +1,338 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { css } from '@emotion/react';
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import {
+  EuiPanel,
+  EuiText,
+  type EuiDataGridCustomBodyProps,
+  useEuiTheme,
+  type UseEuiTheme,
+} from '@elastic/eui';
+import { useVirtualizer } from '@tanstack/react-virtual';
+import {
+  getRenderCustomToolbarWithElements,
+  UnifiedDataTable,
+  DataLoadingState,
+  DataGridDensity,
+  type UnifiedDataTableProps,
+} from '@kbn/unified-data-table';
+import type { CascadeRowCellNestedVirtualizationAnchorProps } from '@kbn/shared-ux-document-data-cascade';
+import type { DataTableRecord } from '@kbn/discover-utils';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+interface ESQLDataCascadeLeafCellProps
+  extends Omit<
+      UnifiedDataTableProps,
+      | 'columns'
+      | 'loadingState'
+      | 'onSetColumns'
+      | 'sampleSizeState'
+      | 'onUpdateSampleSize'
+      | 'onUpdateDataGridDensity'
+      | 'expandedDoc'
+      | 'setExpandedDoc'
+    >,
+    CascadeRowCellNestedVirtualizationAnchorProps<DataTableRecord> {
+  cellData: DataTableRecord[];
+  cellId: string;
+}
+
+interface CustomCascadeGridBodyProps
+  extends EuiDataGridCustomBodyProps,
+    Pick<
+      CascadeRowCellNestedVirtualizationAnchorProps<DataTableRecord>,
+      'getScrollElement' | 'getScrollMargin'
+    > {
+  isFullScreenMode?: boolean;
+  initialOffset: () => number;
+  data: DataTableRecord[];
+}
+
+const getCustomCascadeGridBodyStyle = (euiTheme: UseEuiTheme['euiTheme']) => ({
+  wrapper: css({
+    overflow: 'hidden',
+    position: 'relative',
+    height: '100%',
+    '& .euiDataGridHeader': {
+      backgroundColor: euiTheme.colors.backgroundBaseSubdued,
+    },
+    '& .euiDataGridRow': {
+      backgroundColor: euiTheme.colors.backgroundBaseSubdued,
+    },
+    '.unifiedDataTableToolbar:has(+ .euiDataGrid__content &)': {
+      backgroundColor: euiTheme.colors.backgroundBaseSubdued,
+    },
+    '.unifiedDataTableToolbar:has(+ .euiDataGrid__content &) .unifiedDataTableToolbarControlGroup':
+      {
+        backgroundColor: euiTheme.colors.backgroundBasePlain,
+      },
+  }),
+  headerRow: css({
+    position: 'sticky',
+    top: 0,
+  }),
```

Comment:
Is this supposed to keep the table header pinned to the top when scrolling? If so, it doesn't seem to work.